src/cpp/core/FilePath.cpp

```diff
@@ -471,6 +471,7 @@ MimeType s_mimeTypes[] =
    { "Rprofile", "text/x-r-source"},
    { "Renviron", "text/x-shell" },
    { "rprofvis",   "text/x-r-profile" },
+   { "nb.html", "text/x-r-notebook" },
```

Comment:
I don't think the double-extension will work here either (as we match on just what's after the .nb). I also don't think it's at all problematic to treat the mime type of .nb.html as HTML (will be more broadly compatible).